src/utils/processEvery.ts

```diff
@@ -0,0 +1,16 @@
+import * as humanInterval from 'human-interval';
+import { isValidHumanInterval } from './nextRunAt';
+
+export function calculateProcessEvery(input?: number | string): number {
+	let result = 5000;
```

Comment:
I assume result is the default value, to make this easier to read I would recommend using default value in the parameters.

```ts
export function calculateProcessEvery(input: number | string = 5000): number {
 if (typeof input === 'number') return input;
 return humanInterval(input) as number;
}
```

I guess we can spare the "isVAlidHumanInterval" then.. or we leave this check and throw an error if both cases didn't work out (but I believe this could kinda too over engineered then ;))

Reply:
isn't humanInterval returning a number anyway? is the "as number" required?

Reply:
humanInterval could return undefined if input is not a valid value for humanInterval. So thats why i use the isValidHumanInterval typeguard. As setting the processEvery is usually only set on startup of agenda, i think it is not a really performance issue, either.

We should imho not throw an Error, as the current behaviour of agendaJS is also a fallback to the standard 5 seconds if an invalid value is set. 

This was the first time I wrote a typeguard like isValidHumanInterval, as I usually dont have the opportunity for those :D. 

But if we want to reduce the complexity I guess, we could do it like this:

```typescript
export function calculateProcessEvery(input: number | string = 5000): number {
 if (typeof input === 'number') return input;
 return humanInterval(input) as number || 5000;
}
```

What do you prefer? Keep it like it is? Change?

Reply:
Thanks for the details, yeah let's go with your (shorter) suggestion then :-)